packages/metals-vscode/src/extension.ts

```diff
@@ -195,22 +187,24 @@ function fetchAndLaunchMetals(
   /* eslint-enable @typescript-eslint/no-non-null-assertion */
 
   const coursierMirror = getCoursierMirrorPath(config);
+  const { coursier, javaHome } = await metalsLanguageClient.setupCoursier(
+    javaVersion,
+    context.extensionPath,
```

Comment:
Btw. I think this changes with every version, we should most likely put coursier somewhere where we would not need to redownload with every extension version change